### PR TITLE
With DMD use cmath instead of std.math when std.math widens floats

### PR DESCRIPTION
**Pull request by n8sh**
Only do this when the cmath version is pure. Also support DMD versions prior to 2.082 in mir.math.common.

### CODE REVIEW

File: source/mir/math/common.d [hunk 1]
```diff
@@ -317,43 +346,54 @@ else
     ///
     T cos(T)(in T x) if (isFloatingPoint!T) { return std.math.cos(x); }
     ///
-    T pow(T)(in T x, in T power) if (isFloatingPoint!T) { return std.math.pow(x, power); }
+    T pow(T)(in T x, in T power) if (isFloatingPoint!T) { alias y = power; mixin(mixinCMath2!`pow`); }
     ///
-    T powi(T)(in T x, int power) if (isFloatingPoint!T) { return std.math.pow(x, power); }
+    alias powi = pow;
```
Comment (9il) [inline source/mir/math/common.d]: powi has the restriction for pow type

File: source/mir/math/common.d [hunk 2]
```diff
@@ -370,6 +410,114 @@ else
...
+    ///
+    T sqrt(T)(in T x) if (isFloatingPoint!T) { return std.math.sqrt(x); }
+    ///
+    T sin(T)(in T x) if (isFloatingPoint!T) { mixin(mixinCMath!`sin`); }
+    ///
+    T cos(T)(in T x) if (isFloatingPoint!T) { mixin(mixinCMath!`cos`); }
+    ///
+    T pow(T)(in T x, in T power) if (isFloatingPoint!T) { alias y = power; mixin(mixinCMath2!`pow`); }
+    ///
+    alias powi = pow;
```
Comment (9il) [inline source/mir/math/common.d]: ditto

**Comment by n8sh** [comment]
Made requested changes to `powi`

**Comment by 9il** [comment]
Thanks, feel free to tag a new minor release if required

**Pull request merged**